code/modules/projectiles/projectile/bullets/paintball.dm

```diff
@@ -0,0 +1,46 @@
+/obj/projectile/paintball
+	name = "empty paintball"
+	icon_state = "paintball"
+	damage = 0
+	hitsound = 'sound/weapons/throwtap.ogg'
+
+/obj/projectile/paintball/Initialize(mapload)
+	. = ..()
+	create_reagents(10, NO_REACT)
+
+/obj/projectile/paintball/on_hit(atom/target, blocked = 0, pierce_hit)
+	if(iscarbon(target))
+		var/mob/living/carbon/M = target
+		if(blocked != 100) // not completely blocked
+			..()
+			reagents.trans_to(M, reagents.total_volume, methods = VAPOR)
+			return BULLET_ACT_HIT
```

Comment:
1. Better var name for M, but also
2. Why do we limit this to carbons? Any living mobs can have reagents exposed to them, but also
3. Why limit this to living mobs at all? 

We can just do `reagents.expose(target, VAPOR|TOUCH)`. this is a bit more "proper" than doing a direct transfer

It also has the added bonus of making it so if you miss someone, you'll paint the wall red or blue (which i think is cool). 

For bonus points you can scale the volume the target is exposed to by the `blocked` value, ie `volume_modifier = (blocked / 100)`, so mobs which block a partial amount aren't exposed to all of it.